packages/@react-aria/calendar/src/utils.ts

```diff
@@ -67,10 +76,43 @@ export function useVisibleRangeDescription(startDate: CalendarDate, endDate: Cal
       if (isSameDay(endDate, endOfMonth(startDate))) {
         return monthFormatter.format(startDate.toDate(timeZone));
       } else if (isSameDay(endDate, endOfMonth(endDate))) {
-        return monthFormatter.formatRange(startDate.toDate(timeZone), endDate.toDate(timeZone));
+        return isAria
+          ? formatRange(monthFormatter, formatMessage, startDate, endDate, timeZone)
+          : monthFormatter.formatRange(startDate.toDate(timeZone), endDate.toDate(timeZone));
       }
     }
 
-    return dateFormatter.formatRange(startDate.toDate(timeZone), endDate.toDate(timeZone));
-  }, [startDate, endDate, monthFormatter, dateFormatter, timeZone]);
+    return isAria
+      ? formatRange(dateFormatter, formatMessage, startDate, endDate, timeZone)
+      : dateFormatter.formatRange(startDate.toDate(timeZone), endDate.toDate(timeZone));
+  }, [startDate, endDate, monthFormatter, dateFormatter, formatMessage, timeZone, isAria]);
+}
+
```

Comment:
Testing following story: https://reactspectrum.blob.core.windows.net/reactspectrum/bb967ae0597dcc5288297595e75e08e8d26b9d58/storybook/index.html?path=/story/date-and-time-daterangepicker--hourcycle-12

I notice that without explicitly setting the timezone for the start and end date and time, the the string output within the `aria-describedby` description for each time is 4 hours off. For example, the description reads “Selected Range: April 13, 2022, 5:00 PM to April 15, 2022, 6:00 PM”, when the range selected is “April 13, 2022, 1:00 PM to April 15, 2022, 2:00 PM.”